src/typecheck.c

```diff
@@ -794,17 +794,20 @@ static const Type *typecheck_function_or_method_call(FileTypes *ft, const AstCal
             self_type->name, call->calledname);
     }
 
-    char *sigstr = signature_to_string(sig, false);
+    char *sigstr = signature_to_string(sig, false, false);
 
     int n = call->nargs + !!self_type;
+    int nargs = sig->nargs;
+    if (self_type)
+        nargs--;
     if (n < sig->nargs || (n > sig->nargs && !sig->takes_varargs)) {
```

Comment:
I think we can rewrite this line with `nargs` instead of `n`:
```suggestion
    if (call->nargs < nargs || (call->nargs > nargs && !sig->takes_varargs)) {
```
This way, we no longer need the `n` variable defined above.